src/renderer/components/+preferences/add-helm-repo-dialog.tsx

```diff
@@ -0,0 +1,188 @@
+import "./add-helm-repo-dialog.scss"
+
+import React from "react";
+import { remote } from "electron";
+import { action, observable, runInAction } from "mobx";
+import { observer } from "mobx-react";
+import { t, Trans } from "@lingui/macro";
+import { _i18n } from "../../i18n";
+import { Dialog, DialogProps } from "../dialog";
+import { Wizard, WizardStep } from "../wizard";
+import { Input } from "../input";
+import { Checkbox } from "../checkbox";
+import { systemName, isUrl, isPath } from "../input/input_validators";
+import { SubTitle } from "../layout/sub-title";
+import { Icon } from "../icon";
+import { Notifications } from "../notifications";
+import { HelmRepo, repoManager } from "../../../main/helm/helm-repo-manager";
+
+interface Props extends Partial<DialogProps> {
+  onAddRepo: Function
+}
+
+enum FileType{
+  CaFile,
+  KeyFile,
+  CertFlie,
+}
+
+@observer
+export class AddHelmRepoDialog extends React.Component<Props> {
+  @observable static isOpen = false;
+
+  static open() {
+    AddHelmRepoDialog.isOpen = true;
+  }
+
+  static close() {
+    AddHelmRepoDialog.isOpen = false;
+  }
+
+  @observable helmRepo : HelmRepo = {name: "", url: "", username: "", password: "", insecureSkipTlsVerify: false, caFile:"", keyFile: "", certFile: ""}
+  @observable showOptions = false
+
+  close = () => {
+    AddHelmRepoDialog.close();
+    this.helmRepo = {name: "", url: "", username: "", password: "", insecureSkipTlsVerify: false, caFile:"", keyFile: "", certFile: ""}
```

Comment:
We can move this initial object into private variable to avoid code duplication.